processing/src/main/java/org/apache/druid/query/QueryMetrics.java

```diff
@@ -92,7 +94,7 @@
  * 100% guarantee of compatibility, because methods could not only be added to QueryMetrics, existing methods could also
  * be changed or removed.
  *
- * <p>QueryMetrics is designed for use from a single thread, implementations shouldn't care about thread-safety.
+ * <p><b>All implementations of QueryMetrics should be Thread-safe.</b>
  *
  *
  * Adding new methods to QueryMetrics
```

Comment:
Since the requirements for all methods has changed, probably the name of this class should be changed to "ConcurrentQueryMetrics" to intentionally break proprietary implementations and force them to overhaul.

"DefaultQueryMetrics" OTOH I think could retain the name because it implemented thread-safety itself.

Reply:
Change integrity is more important. I would be understandable if the PR was of thousands of lines already, but it's much smaller and will still be relatively small after the rename.